test/process_managers/process_manager_telemetry_test.exs

```diff
@@ -0,0 +1,316 @@
+
+      assert_receive {[:commanded, :process_manager, :handle, :start], 1, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :stop], 2, measurements, metadata}
+
+      assert match?(%{duration: _}, measurements)
+      assert is_integer(measurements.duration)
+
+      assert match?(
+               %{
+                 application: App,
+                 process_manager_module: ExamplePM,
+                 process_manager_name: "ExamplePM",
+                 process_state: %ExamplePM{message: "init"},
+                 process_uuid: ^uuid,
+                 recorded_event: ^event,
+                 error: :retry
+               },
+               metadata
+             )
+
+      refute_receive {[:commanded, :process_manager, :handle, :exception], _num, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :start], 3, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :stop], 4, _measurements, _metadata}
+    end
+
+    @tag capture_log: true
+    test "emit `[:commanded, :process_manager, :handle, :exception]` event" do
+      uuid = UUID.uuid4()
+
+      {:ok, instance} = start_process_manager_instance(uuid)
+
+      event = to_recorded_event(%Event{message: "exception", type: "raise"})
+
+      :ok = ProcessManagerInstance.process_event(instance, event)
+
+      assert_receive {[:commanded, :process_manager, :handle, :start], 1, _measurements,
+                      _metadata}
+
+      refute_receive {[:commanded, :process_manager, :handle, :stop], _num, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :exception], 2, measurements,
+                      metadata}
+
+      assert match?(%{duration: _}, measurements)
+      assert is_integer(measurements.duration)
+
+      assert match?(
+               %{
+                 application: App,
+                 process_manager_module: ExamplePM,
+                 process_manager_name: "ExamplePM",
+                 process_state: %ExamplePM{message: "init"},
+                 process_uuid: ^uuid,
+                 recorded_event: ^event,
+                 kind: :error,
+                 reason: %RuntimeError{message: "exception"},
+                 stacktrace: _
+               },
+               metadata
+             )
+    end
+  end
+
+  defp attach_telemetry do
+    agent = start_supervised!({Agent, fn -> 1 end})
+
+    :telemetry.attach_many(
+      @handler,
+      [
+        [:commanded, :process_manager, :handle, :start],
+        [:commanded, :process_manager, :handle, :stop],
+        [:commanded, :process_manager, :handle, :exception]
+      ],
+      fn event_name, measurements, metadata, reply_to ->
+        num = Agent.get_and_update(agent, fn num -> {num, num + 1} end)
```

Comment:
I decorated with an incrementing number here so i could make sure the `assert_receive`'s were being received in the right order. Without this you don't get that guarantee, so the tests were misleading